public_html/wp-content/themes/pattern-directory/src/components/pattern-grid/index.js

```diff
@@ -25,8 +25,9 @@ function PatternGrid( { query, children } ) {
 
 	return (
 		<>
+			{ posts.length ? before : null }
```

Comment:
I'm not opposed to this but it does 'lock-in' functionality. If we wanted to show a section with a before and a "no items" view we couldn't (although that would also require a small refactor). Have you considered any other approaches?

Reply:
> If we wanted to show a section with a before and a "no items" view we couldn't

I think that would be a different prop - like, `beforeNone` or something. If we wanted to show something before any content, we could just put it in the parent, like we do in Patterns & MyPatterns, like so:

```
<Menu />
<PatternGrid query={ modifiedQuery }>
	{ ( post ) => <PatternThumbnail key={ post.id } pattern={ post } /> }
</PatternGrid>
```

The point of `before` is to only show when there is content.

Reply:
I typically find components that have a very small number of props easier to modify and make components easier to understand. I can see how a `before` pattern leads to us then having an `after` prop, and so on.... and then we're stuck toggling on/off UI views in a component that should just be a grid of patterns.

Reply:
Would it change anything if the prop was called `header` instead of `before`? I was aiming to keep the `Pattern` component separate from `PatternGrid`, and not fetch the by-author query in both places. I didn't want to overcomplicate `Pattern` by introducing `getPatternsByQuery` & `isLoadingPatternsByQuery`, just to do something that's easily done in `PatternGrid`.

Reply:
Right, I understand the compromise. I think `header` is better than before. 👍